src/utils/calls/estimateGas.ts

```diff
@@ -0,0 +1,39 @@
+import { ethers, Contract } from 'ethers'
+
+/**
+ * Estimate the gas needed to call a function, and add a 10% margin
+ * @param contract Used to perform the call
+ * @param methodName The name of the methode called
+ * @param args An array of arguments to pass to the method
+ * @returns https://docs.ethers.io/v5/api/providers/types/#providers-TransactionReceipt
+ */
+export const estimateGas = async (contract: Contract, methodName: string, methodArgs: any[]) => {
+  if (!contract[methodName]) {
+    throw new Error(`Method ${methodName} doesn't exist on ${contract.address}`)
+  }
+  const rawGasEstimation = await contract.estimateGas[methodName](...methodArgs)
+  // By convention, ethers.BigNumber values are multiplied by 1000 to avoid dealing with real numbers
+  const gasEstimation = rawGasEstimation
+    .mul(ethers.BigNumber.from(10000).add(ethers.BigNumber.from(1000)))
+    .div(ethers.BigNumber.from(10000))
+  return gasEstimation
```

Comment:
Didn't you do this in another PR?

Reply:
just moving to s separate file

Reply:
![image](https://user-images.githubusercontent.com/72658581/123913489-3cf81300-d9b9-11eb-8fb8-511648ef8262.png)